src/scripts/rhsm/download_file_and_report_progress.py

```diff
@@ -0,0 +1,36 @@
+import requests
+import sys
+import json
+import os
+from pathlib import Path
+
+args = json.loads(sys.argv[1], strict=False)
+access_token = args["accessToken"]
+url = args["url"]
+filename = args["fileName"]
+download_dir = args["downloadDir"]
+
+if not os.path.exists(download_dir + filename):
+    # Create download directory, don't complain when it already exists
+    path = Path(download_dir)
+    path.mkdir(parents=True, exist_ok=True)
+
+    # Handle HTTP request failure
+    try:
+        r = requests.get(url, headers={"Authorization": f"Bearer {access_token}"}, stream=True)
+        r.raise_for_status()
+    except requests.exceptions.RequestException as error:
+        sys.exit(error)
+
+    with open(download_dir + filename, "wb") as f:
+        downloaded_length = 0
+        total_length = int(r.headers.get("content-length"))
+        done = -1
+        for data in r.iter_content(chunk_size=4096):
+            downloaded_length += len(data)
+            f.write(data)
+            new_done = int(downloaded_length / total_length * 100)
+            if (new_done > done):
+                done = new_done
+                sys.stdout.flush()
+                sys.stdout.write(f"{done}")
```

Comment:
Why would we need a space separators if we flush? This just prints a single string number as a percentage

Reply:
I meant that you'd print a string like "15812" with this, instead of "1 5 8 12" or even "1\n5\n8\n12". flush() doesn't insert any spaces, it just makes sure an 'incomplete' line gets actually printed.

Reply:
Yeah but the output this script produces isn't meant to be looked at once, since rather check the progess of the download with [process.stream()](https://cockpit-project.org/guide/latest/cockpit-spawn.html#cockpit-spawn-stream), which calls a handler for each single write to a stdout. So it's not really important if I add any whitespaces after percentage, no?

Reply:
It is because you can't guarantee that every write() will have exactly one corresponding read(). The client may be busy with something else for a bit, and then read() the contents from two write()s. Or the other way round, and only read half of the pipe contents. You really need to read(), split on a separator (space or `\n`), and show the last *complete* data set.

With this "block" wise read/write, you need a SEQPACKET socket, not a TCP one or a standard Unix pipe.